File: app/src/main/java/me/weicools/material/component/module/adapter/XFragmentPagerAdapter.java
package me.weicools.material.component.module.adapter;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;
import java.util.List;

/**
 * @author Weicools Create on 2019.03.29
 *
 * desc:
 */
public class XFragmentPagerAdapter extends FragmentPagerAdapter {
  @NonNull private List<Fragment> mFragmentList;
  @Nullable private String[] mTitles;

  public XFragmentPagerAdapter (FragmentManager fm, List<Fragment> fragmentList) {
    super(fm);
    mFragmentList = fragmentList;
  }

  public XFragmentPagerAdapter (FragmentManager fm, List<Fragment> fragmentList, String[] titles) {
    super(fm);
    mTitles = titles;
    mFragmentList = fragmentList;
  }

  @Override
  public Fragment getItem (int position) {
    return mFragmentList.get(position);
  }

  @Override
  public int getCount () {
    return mFragmentList.size();
  }

  @Nullable
  @Override
  public CharSequence getPageTitle (int position) {
    if (mTitles == null || mTitles.length != mFragmentList.size()) {
      return "";
    }
    return mTitles[position];
  }
}

File: app/src/main/java/me/weicools/material/component/common/adapter/XFragmentPagerAdapter.kt
package me.weicools.material.component.common.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

/**
 * @author Weicools Create on 2019.03.29
 *
 * desc:
 */
class XFragmentPagerAdapter : FragmentPagerAdapter {
  private var mFragmentList: List<Fragment>
  private var mTitles: Array<String>? = null

  constructor(fm: FragmentManager, fragmentList: List<Fragment>) : super(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {
    mFragmentList = fragmentList
  }

  constructor(fm: FragmentManager, fragmentList: List<Fragment>, titles: Array<String>) : super(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {
    mTitles = titles
    mFragmentList = fragmentList
  }

  override fun getItem(position: Int): Fragment {
    return mFragmentList[position]
  }

  override fun getCount(): Int {
    return mFragmentList.size
  }

  override fun getPageTitle(position: Int): CharSequence? {
    return if (mTitles == null || mTitles!!.size != mFragmentList.size) {
      ""
    } else mTitles!![position]
  }
}
